scripts/research_issue_lint.py

```
@@ -0,0 +1,171 @@
+        entry_id = entry.get("id")
+        if not isinstance(entry_id, str) or not re.match(r"^[a-z0-9][a-z0-9_-]*$", entry_id):
+            errors.append(f"{path}:{line_number}: invalid id")
+        elif entry_id in seen_ids:
+            errors.append(f"{path}:{line_number}: duplicate id {entry_id}")
+        else:
+            seen_ids.add(entry_id)
+        if entry.get("status") not in LEDGER_STATUSES:
+            errors.append(f"{path}:{line_number}: invalid status {entry.get('status')!r}")
+        for field in ("required_artifacts", "non_claims"):
+            if not isinstance(entry.get(field), list) or not entry.get(field):
+                errors.append(f"{path}:{line_number}: {field} must be a non-empty list")
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)